File: JAVA/ValidSudoku.java
public class ValidSudoku {
    
    public boolean isValidSudoku(char[][] board) {
        //The Java.util.HashSet.add() method will add the element only if the specified element is not present in the HashSet else the function will return False if the element is already present in the HashSet
        HashSet<String> set = new HashSet<>();
       // we iterate throught the board only one time and store the strings and if we find any of the string already present in
        // the hashset again then we return false
      for(int i=0;i<9; i++)
      {
          for(int j=0;j<9;j++)
          {
              char ch = board[i][j];
              String s ="";
              if(ch != '.')
              {
                 if(!set.add(s + "row" + i + ch) || !set.add(s + "column" + j + ch)) 
                 {
                     
                     return false;
                 }
                  // (row/3 * 3 + column/3 is a formula to get the box no , just a way to maintain consistent box number,  dry run this formula for 
				  // better  understanding
                  if(!set.add(s + "box" + (((i/3) *3) + j/3) + ch))
                      return false;
              }
          }
      }
        
        return true;
      
    }
   
}

File: Leetcode/valid_sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& b) {
        for(int i=0;i<9;i++){
            vector<int> v(10,0);
            for(int j=0;j<9;j++){
                if(b[i][j]=='.')    continue;
                v[b[i][j]-'0']++;
                if(v[b[i][j]-'0']>1)    return false;
            }
        }
        for(int i=0;i<9;i++){
            vector<int> v(10,0);
            for(int j=0;j<9;j++){
                if(b[j][i]=='.')    continue;
                v[b[j][i]-'0']++;
                if(v[b[j][i]-'0']>1)    return false;
            }
        }
        vector<vector<vector<int>>> v(3,vector<vector<int>> (3,vector<int>(10,0)));
        for(int i=0;i<9;i++){
            for(int j=0;j<9;j++){
                if(b[i][j]=='.')    continue;
                int r,c;
                r=i/3;
                c=j/3;
                v[r][c][b[i][j]-'0']++;
                if(v[r][c][b[i][j]-'0']>1)  return false;
            }
        }
        return true;
    }
};